splunk_topology/check.py

```diff
@@ -123,10 +123,10 @@ def check(self, instance):
             if instance.snapshot:
                 self.stop_snapshot(instance_key)
         except Exception as e:
-            self._clear_topology(instance_key, clear_in_snapshot=True)
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=instance.tags, message=str(e))
             self.log.exception("Splunk topology exception: %s" % str(e))
             if not instance.splunk_ignore_saved_search_errors:
+                self._clear_topology(instance_key, clear_in_snapshot=True)
                 raise CheckException("Splunk topology failed with message: %s" % e), None, sys.exc_info()[2]
             self.log.warning("Ignoring Splunk topology exception as ignore_saved_search_errors flag is true.")
```

Comment:
this case should never be hit, beacuase the issue is now caught on the lower level, so we can ditch this part

Reply:
Done.